tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQHeadersCollectionAdapter.cs

```diff
@@ -0,0 +1,43 @@
+// <copyright file="RabbitMQHeadersCollectionAdapter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ
+{
+    internal readonly struct RabbitMQHeadersCollectionAdapter : IBinaryHeadersCollection
+    {
+        private static readonly byte[] EmptyValue = Array.Empty<byte>();
+        private readonly IDictionary<string, object> _headers;
+
+        public RabbitMQHeadersCollectionAdapter(IDictionary<string, object> headers)
+        {
+            _headers = headers;
+        }
+
+        public byte[] TryGetLastBytes(string name)
+        {
+            if (_headers != null && _headers.TryGetValue(name, out var value) && value is byte[] bytes)
```

Comment:
With the nullable reference types feature enabled (in line 5 `#nullable enable`) and null-checks before entering this method, I think we can safely remove all the null checks on the `_headers` field